conans/cli/command.py

```diff
@@ -117,17 +117,17 @@ def add_subcommand(self, subcommand):
         subcommand.set_parser(self._parser, self._subcommand_parser)
         self._subcommands[subcommand.name] = subcommand
 
-    def run(self, *args, conan_api, **kwargs):
-        info = self._method(*args, conan_api=conan_api, **kwargs)
+    def run(self, conan_api, cli_out, parser, *args, **kwargs):
+        info = self._method(conan_api, parser, *args, **kwargs)
         if not self._subcommands:
             parser_args = self._parser.parse_args(*args)
             if info:
-                self._formatters[parser_args.output](info, conan_api.out)
+                self._formatters[parser_args.output](info, cli_out)
         else:
             arg_list, = args
             subcommand = arg_list[0]
```

Comment:
Instead of taking the first from `args`, can we add `subcommand=None` to the arguments and avoid these two lines?

Reply:
I'm not sure of that, all the information to decide about the subcommands is inside this class... maybe it is more confusing if we pass a potential subcommand without knowing if that conan command accepts subcommands or not...

Reply:
I understand it, but for me this is like the _"Voy a tener suerte"_ in Google search 😅  I see in the same method a couple of checks like `if not self._subcommands:` and `if subcommand in self._subcommands:`, so we are already trying to check if the command accepts subcommands or not.

IMHO a implementation like this would be more clear:

```python
def run(self, conan_api, cli_out, parser, subcommand=None, *args, **kwargs):

    if subcommand:
        try:
            self._subcommands[subcommand].run(conan_api, cli_out, *args)
        except KeyError:
            # Fallback behavior
    else:
        info = self._method(conan_api, parser, *args, **kwargs)
        parser_args = self._parser.parse_args(*args)
        self._formatters[parser_args.output](info, cli_out)
```

but I'm not running this locally, so this might only work in an ideal world 😅